app/src/test/java/com/malibin/study/github/data/repository/DefaultGithubProfileRepositoryTest.kt

```diff
@@ -0,0 +1,143 @@
+package com.malibin.study.github.data.repository
+
+import com.google.common.truth.Truth.assertThat
+import com.malibin.study.github.data.source.GithubProfileSource
+import com.malibin.study.github.domain.profile.GithubProfile
+import com.malibin.study.github.domain.repository.GithubProfileRepository
+import com.malibin.study.github.utils.InstantTaskExecutorExtension
+import com.malibin.study.github.utils.MainCoroutineExtension
+import io.mockk.coEvery
+import io.mockk.coVerify
+import io.mockk.coVerifySequence
+import io.mockk.impl.annotations.RelaxedMockK
+import io.mockk.junit5.MockKExtension
+import io.mockk.slot
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.runTest
+import org.junit.jupiter.api.Assertions.*
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.RegisterExtension
+
+@OptIn(ExperimentalCoroutinesApi::class)
+@ExtendWith(MockKExtension::class)
+internal class DefaultGithubProfileRepositoryTest {
+
+    @RelaxedMockK
+    lateinit var localGithubProfileSource: GithubProfileSource
+
+    @RelaxedMockK
+    lateinit var remoteGithubProfileSource: GithubProfileSource
+    private lateinit var githubProfileRepository: GithubProfileRepository
+
+    @BeforeEach
+    fun setUp() {
+        githubProfileRepository = DefaultGithubProfileRepository(
+            localGithubProfileSource,
+            remoteGithubProfileSource
+        )
+    }
+
+    @Test
+    fun `githubProfile을 저장할 떄 내부 저장소에 저장된다`() = runTest {
+        // given
+        val profile = GithubProfile(
+            id = 1,
+            userName = "June",
+            avatarUrl = "",
+            name = "",
+            bio = "",
+            followersCount = 11,
+            followingCount = 12
+        )
+        // when
+        val actualResult = githubProfileRepository.saveGithubProfile(profile)
+        // then
+        assertAll(
+            { coVerify(exactly = 1) { localGithubProfileSource.saveGithubProfile(profile) } },
+            { assertThat(actualResult.isSuccess).isTrue() }
+        )
+    }
+
+    @Test
+    fun `local저장소에 userName에 해당하는 profile이 있으면 가져온다`() = runTest {
+        // given
+        val userName = "June"
+        val expectedProfile = GithubProfile(
+            id = 1,
+            userName = "June",
+            avatarUrl = "",
+            name = "",
+            bio = "",
+            followersCount = 11,
+            followingCount = 12
+        )
+        coEvery { githubProfileRepository.getGithubProfile(userName) } returns Result.success(
+            expectedProfile
+        )
```

Comment:
저는 이렇게 정렬하는 게 더 보기 좋더라구요 :)
```suggestion
        coEvery { 
            githubProfileRepository.getGithubProfile(userName) 
        } returns Result.success(expectedProfile)
```